README.md
# local-gov-indicators
**Data Set on Local Government Indicators in Chile**

[![Coverage](https://raw.githubusercontent.com/bgonzalezbustamante/local-gov-indicators/main/badges/coverage_20.svg)](https://github.com/bgonzalezbustamante/local-gov-indicators/tree/main/data/tidy/cases) [![Version](https://raw.githubusercontent.com/bgonzalezbustamante/local-gov-indicators/main/badges/v_0_21_15.svg)](CHANGELOG.md) [![Project Status: Suspended – Initial development has started, but there has not yet been a stable, usable release; work has been stopped for the time being but the author(s) intend on resuming work.](https://raw.githubusercontent.com/bgonzalezbustamante/local-gov-indicators/main/badges/suspended.svg)](STATUS.md) [![DOI](https://raw.githubusercontent.com/bgonzalezbustamante/local-gov-indicators/main/badges/doi.svg)](https://doi.org/10.5281/zenodo.7568387) [![License](https://raw.githubusercontent.com/bgonzalezbustamante/local-gov-indicators/main/badges/gnu.svg)](LICENSE-GPL.md) [![License](https://raw.githubusercontent.com/bgonzalezbustamante/local-gov-indicators/main/badges/cc_by_4_0.svg)](LICENSE-CC.md) [![License](https://raw.githubusercontent.com/bgonzalezbustamante/local-gov-indicators/main/badges/lppl.svg)](LICENSE-LPPL.md) [![R](https://raw.githubusercontent.com/bgonzalezbustamante/local-gov-indicators/main/badges/r_4_1_0.svg)](https://cran.r-project.org/) [![Latex](https://img.shields.io/badge/made%20with-LaTeX-1f425f.svg)](https://www.latex-project.org/) [![tDL](https://raw.githubusercontent.com/bgonzalezbustamante/local-gov-indicators/main/badges/tDL.svg)](https://training-datalab.com/)

## Overview

This repository contains a data set in progress (20%) on local government indicators in Chile between 2010 and 2021, featuring an e-government indicator (EGI) in 2016[^1], 2019 and 2021 in Comma-Separated Values `CSV` format with Unicode encoding `UTF-8`. Part of this data set has been used in the following articles:

- González-Bustamamante, B., & Aguilar, D. (2023, *forthcoming*). Territorial patterns of open e-government: Evidence from Chilean municipalities. *Political Research Exchange*.

## Data Pipeline and Coverage

### Pipeline

```mermaid
  graph TD;
      A[/Shapefiles<br/>IDE-Chile/] --> B[(Baseline<br/>Data Set)];
      B --> C[(Case-Level<br/>Data Sets)];
      D[\Municipalities<br/>Websites/] --> E[(EGI)];
      E[(EGI)] --> C;
      F[/CPLT/] --> G[(TA)];
      F[/CPLT/] --> H[(DAI)];
      F[/CPLT/] --> I[(FOIA)];
      G --> C;
      H --> C;
      I --> C;
      J[/OS-MDS/] --> K[(SAE)];
      K --> C;
      L[/SINIM/] --> C;
      M[/INE/] --> C;
      N[/SUBTEL/] --> C;
      C --> V[Validation];
      V --> C;
      C --> Z[(Case-Level<br/>Time-Series<br/>Data Set)];
```
### Coverage

|  | EGI | TA | DAI | SAE | SINIM | INE | SUBTEL |
|---|:---:|:---:|:---:|:---:|:---:|:---:|---|
| 2010 | :x: | :x: | :x: | :x: |  |  |  |
| 2011 | :x: | :x: | :x: | :white_check_mark: |  |  |  |
| 2012 | :x: | :white_check_mark: | :x: | :x: |  |  |  |
| 2013 | :x: | :white_check_mark: | :x: | :white_check_mark: |  |  |  |
| 2014 | :x: | :white_check_mark: | :white_check_mark: | :x: |  |  |  |
| 2015 | :x: | :white_check_mark: | :white_check_mark: | :white_check_mark: |  |  |  |
| 2016 | :white_check_mark: | :white_check_mark: | :white_check_mark: | :x: |  |  |  |
| 2017 | :x: | :white_check_mark: | :white_check_mark: | :white_check_mark: |  |  |  |
| 2018 | :x: | :x: | :white_check_mark: | :x: |  |  |  |
| 2019 | :white_check_mark: | :x: | :x: | :x: |  |  |  |
| 2020 | :x: | :x: | :x: | :white_check_mark: |  |  |  |
| 2021 | :white_check_mark: | :x: | :x: | :x: |  |  |  |

## Preservation

This data set is stored with version control on a GitHub repository. Furthermore, a Digital Object Identifier is provided by Zenodo.

## Storage and Backup

The GitHub repository has controlled access with Two-Factor Authentication `2FA` with two physical USB security devices (Bastián González-Bustamante, [ORCID iD 0000-0003-1510-6820](https://orcid.org/0000-0003-1510-6820)) and a mobile application (Diego Aguilar, [ORCID iD 0000-0003-4531-5922](https://orcid.org/0000-0003-4531-5922)). USB devices and the mobile application issue one-time passwords to generate a cryptographic authentication `FIDO2` and `U2F`.

Moreover, the repository is backed up on Hierarchical File Server `HFS` for recovery in case of incidents. This backup is located on the University of Oxford hub connected to `Code42` Cloud Backup encrypted with `256-bit AES`. The backup is performed with every change on GitHub and receives weekly light maintenance and a deep one every month. This backup will be secured until May 2024. An extension of this period will be evaluated on budget availability.

## Getting Started

### Software

We used `R v4.1.0 -- Camp Pontanezen`.

## License

The content of this project itself is licensed under a [Creative Commons Attribution 4.0 International license (CC BY 4.0)](LICENSE-CC.md), and the underlying code used to format and display that content is licensed under a [GNU GPLv3 license](LICENSE-GPL.md). In particular, the codebook will be compiled with XeLaTeX using a code licensed under an [LPPL v1.3c license](LICENSE-LPPL.md).

The above implies that the data sets may be shared, reused, and adapted as long as appropriate acknowledgement is given. In addition, the code may be shared, reused, and adapted as long as the source is disclosed, the changes are stated, and the same [GNU GPLv3 license](LICENSE-GPL.md) is used.

## Contribute

Contributions are entirely welcome. You just need to [open an issue](https://github.com/bgonzalezbustamante/local-gov-indicators/issues/new) with your comment or idea.

For more substantial contributions, please fork this repository and make changes. Pull requests are also welcome.

Please read our [code of conduct](CODE_OF_CONDUCT.md) first. Minor contributions will be acknowledged, and significant ones will be considered on our contributor roles taxonomy.

## Citation

González-Bustamante, B., & Aguilar, D. (2023). Data Set on Local Government Indicators in Chile (Version 0.21.15 -- Late Sky) [Data set, pre-release version]. University of Oxford, Universidad de Santiago de Chile (USACH) and Training Data Lab. DOI: [10.5281/zenodo.7568387](https://doi.org/10.5281/zenodo.7568387)

## Authors

Bastián González-Bustamante \
bastian.gonzalezbustamante@politics.ox.ac.uk \
[ORCID iD 0000-0003-1510-6820](https://orcid.org/0000-0003-1510-6820) \
https://bgonzalezbustamante.com

Diego Aguilar \
diego.aguilar@training-datalab.com \
[ORCID iD 0000-0003-4531-5922](https://orcid.org/0000-0003-4531-5922)

## CRediT - Contributor Roles Taxonomy

Bastián González-Bustamante ([ORCID iD 0000-0003-1510-6820](https://orcid.org/0000-0003-1510-6820)): Conceptualisation, data curation, formal analysis, funding acquisition, methodology, project administration, resources, software, supervision, and validation.

Diego Aguilar ([ORCID iD 0000-0003-4531-5922](https://orcid.org/0000-0003-4531-5922)): Conceptualisation, data curation, investigation, resources, and validation.

Berenice Orvenes ([OSF osf.io/37keu](https://osf.io/37keu/)): Investigation and resources.

Ariane Carvajal ([ORCID iD 0000-0002-6181-7458](https://orcid.org/0000-0002-6181-7458)): Investigation and resources.

Andrea González ([ORCID iD 0000-0002-6822-0056](https://orcid.org/0000-0002-6822-0056)): Investigation and resources.

Elinor Luco ([ORCID iD 0000-0003-3370-4322](https://orcid.org/0000-0003-3370-4322)): Investigation.

### Latest Revision

[March 17, 2023](CHANGELOG.md).

[^1]: Further details of the 2016 indicator in González-Bustamante, B., Carvajal, A., & González, A. (2020). Determinantes del gobierno electrónico en las municipalidades: Evidencia del caso chileno. *Gestión y Política Pública, XXIX*(1), 97-129. DOI: [10.29265/gypp.v29i1.658](http://dx.doi.org/10.29265/gypp.v29i1.658). See also González-Bustamante, B., Carvajal, A., & González, A. (2019). Local E-Government Index 2016 in Chile. DOI: [10.7910/DVN/CMU36B](https://doi.org/10.7910/DVN/CMU36B).


CHANGELOG.md
# Changelog
All notable changes to "Data Set on Local Government Indicators in Chile" will be documented in this file.

## [Unreleased]

## [1.0.0] - TBC
### Added
- Codebook in XeLaTeX.
- Release of VOR data set.
- Exploratory data analysis code.
- Import data code.
- BibTeX file.

## [Released]

## [0.21.15] - 2023-03-17
### Changed
- Badges in SVG on GitHub repository.
- Minor corrections.

## [0.21.14] - 2023-01-25
### Changed
- Badges in SVG on GitHub repository.
- Revision of upgrade versioning.
- Minor corrections.
### Removed
- FOIA raw data.
- Scatters EGI indexes.

## [0.21.13] - 2022-07-17
### Added
- Zenodo DOI.
### Changed
- Minor corrections.
### Deprecated
- FOIA raw data.
- Scatters EGI indexes.

## [0.20.13] - 2022-07-09
### Changed
- SUBTEL raw data sets.
- SINIM raw data sets.

## [0.20.12] - 2022-07-08
### Changed
- SUBTEL raw data sets.
- SAE estimations.
- Data cleaning code.
- Revision of upgrade versioning.
- Minor corrections.

## [0.20.11] - 2022-07-01
### Changed
- SINIM raw data sets.
- Revision of upgrade versioning.
- Minor corrections.

## [0.20.10] - 2022-05-26
### Changed
- Minor corrections.

## [0.20.9] - 2022-05-06
### Added
- SUBTEL raw data sets.
### Changed
- Data cleaning code.
- Minor corrections.

## [0.19.9] - 2022-04-19
### Changed
- Minor corrections.

## [0.19.8] - 2022-04-12
### Added
- FOIA raw data.
- Transparency sources backups on OSF.
### Changed
- CPLT data sets.

## [0.18.8] - 2022-04-06
### Changed
- Minor corrections.

## [0.18.7] - 2022-04-03
### Changed
- Minor corrections.

## [0.18.6] - 2022-04-01
### Changed
- Minor corrections.

## [0.18.5] - 2022-03-31
### Added
- GNU GPLv3 license on GitHub repository.
- LPPL v1.3c license on GitHub repository.
### Changed
- Minor corrections.
### Removed
- MIT license on GitHub repository.

## [0.17.5] - 2022-03-25
### Added
- Scatters EGI indexes.
### Changed
- Data cleaning code.
- Minor corrections.
### Deprecated
- MIT license on GitHub repository.

## [0.16.5] - 2022-03-24
### Changed
- Data cleaning code.

## [0.16.4] - 2022-03-22
### Added
- MIT license on GitHub repository.
### Changed
- Revision of upgrade versioning.
- Minor corrections.

## [0.15.4] - 2022-03-22
### Added
- Finishing CUT data revision.

## [0.14.4] - 2022-03-18
### Added
- Git LFS.
- Communes shapefiles.
### Changed
- Modify .gitignore file on GitHub repository.

## [0.13.4] - 2022-03-16
### Added
- Data cleaning code.
- Data pipeline Mermaid diagram in README.
- Data coverage information in README.

## [0.12.4] - 2022-03-15
### Added
- Shapefiles cleaning code.
- CUT data backup from IDE-Chile.
- CUT shapefiles in CSV UTF-8 format.
- Communes shapefiles.
### Changed
- Modify .gitignore file on GitHub repository.
- Minor corrections.

## [0.11.4] - 2022-03-12
### Changed
- Modify .gitignore file on GitHub repository.
- Minor corrections.

## [0.11.3] - 2022-03-10
### Changed
- Repository name.
- Revision of upgrade versioning.

## [0.11.2] - 2022-03-04
### Added
- SAE estimations.

## [0.10.2] - 2022-03-03
### Added
- Starting CUT data revision.

## [0.9.2] - 2022-03-02
### Added
- CPLT data sets.

## [0.8.2] - 2022-02-25
### Added
- SINIM raw data sets.
- SEGPRES digitalisation data set.
### Changed
- Minor corrections.

## [0.7.2] - 2022-02-23
### Added
- Code of conduct in MD format.
- Citation CFF file.
### Changed
- Scripts and folders names standardisation.
- Revision of upgrade versioning.
- Minor corrections.

## [0.6.2] - 2021-11-15
### Added
- Raw EGi 2021 measurement.

## [0.5.2] - 2022-02-05
### Changed
- Minor corrections.

## [0.5.1] - 2021-11-18
### Changed
- Minor corrections.

## [0.5.0] - 2021-11-17
### Added
- Repository status badge.
### Changed
- Minor corrections.

## [0.4.0] - 2021-11-15
### Added
- Badges in SVG on GitHub repository.
### Changed
- License in MD format on GitHub repository.

## [0.3.0] - 2021-11-02
### Added
- Raw EGi 2016 measurement.

## [0.2.0] - 2021-10-31
### Added
- Changelog in MD format on GitHub repository.
- License CC BY 4.0 on GitHub repository.
- Local EGi 2016 data set in DTA format (DOI: 10.7910/DVN/CMU36B).
- Raw EGi 2019 measurement.

## [0.1.0] - 2021-10-28
### Added
- GitHub repository.
- README in MD format on GitHub repository.


code/stage_1_shapefiles_cleaning.R
## Draft Code Shapefiles
## Bastián González-Bustamante
## March 2022

## Clean Environment
rm(list = ls())

## Packages
library(rgdal)
library(tidyverse)

communes <- readOGR("data/raw/shapefiles/COMUNAS_2020.shp", p4s = NULL, verbose = TRUE,
                    use_iconv = TRUE, encoding = "UTF-8")

communes$CUT_REG <- as.numeric(communes$CUT_REG)
communes_01 <- communes[communes$CUT_REG == 1,]
communes_02 <- communes[communes$CUT_REG == 2,]
communes_03 <- communes[communes$CUT_REG == 3,]
communes_04 <- communes[communes$CUT_REG == 4,]
communes_05 <- communes[communes$CUT_REG == 5,]
communes_06 <- communes[communes$CUT_REG == 6,]
communes_07 <- communes[communes$CUT_REG == 7,]
communes_08 <- communes[communes$CUT_REG == 8,]
communes_09 <- communes[communes$CUT_REG == 9,]
communes_10 <- communes[communes$CUT_REG == 10,]
communes_11 <- communes[communes$CUT_REG == 11,]
communes_12 <- communes[communes$CUT_REG == 12,]
communes_13 <- communes[communes$CUT_REG == 13,]
communes_14 <- communes[communes$CUT_REG == 14,]
communes_15 <- communes[communes$CUT_REG == 15,]
communes_16 <- communes[communes$CUT_REG == 16,]

plot(communes_13)

## Dataframe 

cut_shapefiles <- data.frame(CUT_REG = communes$CUT_REG,
                             REGION = communes$REGION,
                             CUT_PROV = communes$CUT_PROV,
                             PROVINCIA = communes$PROVINCIA,
                             CUT_COM = communes$CUT_COM,
                             COMUNA = communes$COMUNA,
                             SUPERFICIE = communes$SUPERFICIE)

## Save File
write.csv(cut_shapefiles, "data/raw/CUT/cut_shapefiles.csv", fileEncoding = "UTF-8", row.names =  FALSE)


code/stage_2_data_cleaning.R
## +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

#### Script ID ####

## Data Cleaning Local Government Indicators
## R version 4.1.0 (2021-05-18) -- "Camp Pontanezen"
## Date: March and July 2022

## Bastián González-Bustamante (University of Oxford, UK)
## https://bgonzalezbustamante.com
## Diego Aguilar (Training Data Lab)
## https://training-datalab.com

## Data Set on Local Government Indicators in Chile
## https://github.com/bgonzalezbustamante/local-gov-indicators

## +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

#### Packages and Data ####

## +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

## Clean Environment
rm(list = ls())

## Packages
library(tidyverse)
library(foreign)
## library(ltm)
library(psych)

## Shapefiles
cut_shapefiles <- read.csv("data/raw/CUT/cut_shapefiles.csv", encoding = "UTF-8")
names(cut_shapefiles)[1] = "cut_reg"
names(cut_shapefiles)[2] = "region"
names(cut_shapefiles)[3] = "cut_prov"
names(cut_shapefiles)[4] = "province"
names(cut_shapefiles)[5] = "cut_com"
names(cut_shapefiles)[6] = "commune"
names(cut_shapefiles)[7] = "area"

## EGI 2016
egi_2016 <- read.dta("data/raw/EGI/EGI-2016-vor-Stata12.dta")
names(egi_2016)[2] = "cut_com"

## TA 2016
ta_2016 <- read.csv("data/raw/CPLT/TA/ta_municipios_2016.csv", encoding = "UTF-8")
names(ta_2016)[1] = "cplt_code"
names(ta_2016)[3] = "ta"

## DAI 2016
dai_2016 <- read.csv("data/raw/CPLT/DAI/dai_municipios_online_2016.csv", encoding = "UTF-8")
names(dai_2016)[2] = "cplt_code"
names(dai_2016)[3] = "dai"

## EGI 2019
egi_2019 <- read.csv("data/raw/EGI/egi_2019.csv", encoding = "UTF-8")
names(egi_2019)[1] = "region"

## SINIM 2019
sinim_2019 <- read.csv("data/raw/SINIM/sinim_2019.csv", encoding = "UTF-8")
names(sinim_2019)[1] = "cut_com"

## SUBTEL 2019
subtel_2019 <- read.csv("data/raw/SUBTEL/data_SUBTEL_2019.csv", encoding = "UTF-8")
names(subtel_2019)[1] = "comuna"
names(subtel_2019)[2] = "cut_com"
names(subtel_2019)[15] = "internet_con"

## EGI 2021
egi_2021 <- read.csv("data/raw/EGI/egi_2021.csv", encoding = "UTF-8")
names(egi_2021)[1] = "commune"
egi_2021 <- filter(egi_2021, cut_com != 12202)

## SINIM 2021
sinim_2021 <- read.csv("data/raw/SINIM/sinim_2021.csv", encoding = "UTF-8")
names(sinim_2021)[1] = "cut_com"

## SUBTEL 2021
subtel_2021 <- read.csv("data/raw/SUBTEL/data_SUBTEL_2021.csv", encoding = "UTF-8")
names(subtel_2021)[1] = "comuna"
names(subtel_2021)[2] = "cut_com"
names(subtel_2021)[15] = "internet_con"

## +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

#### Case Level-Data Set 2016 ####

## +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

## Alpha
alpha_egi_2016 <- select(egi_2016, forms, news, webmap, streetmap, transport, socialmedia, phone, 
                         mobile, transac, followup, citizen, payments, part)
## alpha_egi_2016 <- alpha_egi_2016 %>% 
## filter(if_all(everything(), ~ !is.na(.x)))
## alpha(alpha_egi_2016)
ltm::cronbach.alpha(alpha_egi_2016, CI = TRUE, na.rm = TRUE)

## EGI
egi_2016$egi <- ((egi_2016$forms + egi_2016$news + egi_2016$webmap) * 0.25) + 
  ((egi_2016$streetmap + egi_2016$transport) * 0.50) +
  ((egi_2016$socialmedia + egi_2016$phone + egi_2016$mobile) * 0.75) +
  ((egi_2016$transac + egi_2016$followup + egi_2016$citizen + egi_2016$payments) * 1.00) +
  ((egi_2016$part) * 1.25)
sum(egi_2016$egi, na.rm = TRUE)

## egi_2016$egi[which(egi_2016$commune == "LAS CONDES")]
## egi_2016$egi[which(egi_2016$commune == "LO BARNECHEA")]
## egi_2016$egi[which(egi_2016$commune == "PROVIDENCIA")]

## EGI Std
score_2016 <- (3 * 0.25) + (2 * 0.50) + (3 * 0.75) + (4 * 1.00) + (1 * 1.25)
score_2019 <- (3 * 0.25) + (2 * 0.50) + (3 * 0.75) + (5 * 1.00) + (1 * 1.25)
egi_2016$egi_std <- (egi_2016$egi / score_2016) * score_2019

## Local Gov Indicators
local_gov_indicators_2016 <- cut_shapefiles
local_gov_indicators_2016 <- cbind(year = 2016, local_gov_indicators_2016)
local_gov_indicators_2016 <- cbind(id = rownames(local_gov_indicators_2016), local_gov_indicators_2016)

## Corrections
source("code/functions/cut_new_region.R", encoding = "UTF-8")

## Merge
local_gov_indicators_2016 <- left_join(local_gov_indicators_2016, egi_2016[c("cut_com", "egi", "egi_std")], by = "cut_com")
sum(local_gov_indicators_2016$egi, na.rm = TRUE)

## local_gov_indicators_2016 %>%
  ## group_by(cut_reg) %>%
  ## summarise(mean = mean(egi, na.rm = TRUE), n = n())

## CPLT Data
source("code/functions/cplt_code.R", encoding = "UTF-8")
local_gov_indicators_2016 <- bind_cols(local_gov_indicators_2016, cplt_code)
names(local_gov_indicators_2016)[12] = "cplt_code"
local_gov_indicators_2016 <- left_join(local_gov_indicators_2016, ta_2016[1:3], by = "cplt_code")
local_gov_indicators_2016 <- left_join(local_gov_indicators_2016, dai_2016[1:3], by = "cplt_code")
local_gov_indicators_2016$dai <- local_gov_indicators_2016$dai %>% factor %>% str_replace(',', '.') %>% as.numeric
local_gov_indicators_2016$cplt_code <- NULL
local_gov_indicators_2016$Nombre.x <- NULL
local_gov_indicators_2016$Nombre.y <- NULL

## SAE Data
local_gov_indicators_2016$sae <- NA

## SINIM Data
## Pending

## SUBTEL Data
## Pending

## Case-Level CSV
write.csv(local_gov_indicators_2016, "data/tidy/cases/local_gov_indicators_2016.csv", 
          fileEncoding = "UTF-8", row.names =  FALSE)

## +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

#### Case Level-Data Set 2019 ####

## +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

## Alpha
## table(egi_2019$region)
## Incorporating digital_certificate
subsample_2019 <- filter(egi_2019, region == "COQUIMBO" | region == "VALPARAISO" 
                         | region == "METROPOLITANA" | region == "BIOBIO" | region == "DE LA ARAUCANIA")

alpha_egi_2019 <- select(subsample_2019, forms, news, webmap, streetmap, transport, socialmedia, phone, 
                         mobile, transac, followup, digital_certificate, citizen, payments, part)
## alpha(alpha_egi_2019)
ltm::cronbach.alpha(alpha_egi_2019, CI = TRUE, na.rm = TRUE)

## EGI
egi_2019$egi <- ((egi_2019$forms + egi_2019$news + egi_2019$webmap) * 0.25) + 
  ((egi_2019$streetmap + egi_2019$transport) * 0.50) +
  ((egi_2019$socialmedia + egi_2019$phone + egi_2019$mobile) * 0.75) +
  ((egi_2019$transac + egi_2019$followup + egi_2019$digital_certificate + egi_2019$citizen + egi_2019$payments) * 1.00) +
  ((egi_2019$part) * 1.25)
sum(egi_2019$egi, na.rm = TRUE)

## Local Gov Indicators
local_gov_indicators_2019 <- cut_shapefiles
local_gov_indicators_2019 <- cbind(year = 2019, local_gov_indicators_2019)
local_gov_indicators_2019 <- cbind(id = rownames(local_gov_indicators_2019), local_gov_indicators_2019)

## Corrections
egi_2019$cut_com[which(egi_2019$commune == "MAFIL")] <- local_gov_indicators_2019$cut_com[which(local_gov_indicators_2019$commune == "Máfil")]
egi_2019$cut_com[which(egi_2019$commune == "PUTRE")] <- 15201
egi_2019$cut_com[which(egi_2019$commune == "GENERAL LAGOS")] <- 15202
egi_2019$cut_com[which(egi_2019$commune == "RANQUIL")] <- local_gov_indicators_2019$cut_com[which(local_gov_indicators_2019$commune == "Ranquil")]

## Merge
local_gov_indicators_2019 <- left_join(local_gov_indicators_2019, egi_2019[c("cut_com", "egi")], by = "cut_com")
sum(local_gov_indicators_2019$egi, na.rm = TRUE)
local_gov_indicators_2019$egi_std <- NA

## CPLT Data
local_gov_indicators_2019$ta <- NA
local_gov_indicators_2019$dai <- NA

## SAE Data
local_gov_indicators_2019$sae <- NA

## SINIM Data
local_gov_indicators_2019 <- left_join(local_gov_indicators_2019, sinim_2019, by = "cut_com")
local_gov_indicators_2019$municipality <- NULL
local_gov_indicators_2019$iadm40 <- as.numeric(local_gov_indicators_2019$iadm40)
local_gov_indicators_2019$iadm44 <- as.numeric(local_gov_indicators_2019$iadm44)
local_gov_indicators_2019$iadm41 <- as.numeric(local_gov_indicators_2019$iadm41)
local_gov_indicators_2019$iadm25 <- as.numeric(local_gov_indicators_2019$iadm25)
local_gov_indicators_2019$iadm33 <- as.numeric(local_gov_indicators_2019$iadm33)
local_gov_indicators_2019$isoc013 <- as.numeric(local_gov_indicators_2019$isoc013)
local_gov_indicators_2019$mocosoc <- as.factor(local_gov_indicators_2019$mocosoc)
local_gov_indicators_2019$mcosoc <- as.factor(local_gov_indicators_2019$mcosoc)
local_gov_indicators_2019$icar007 <- as.numeric(local_gov_indicators_2019$icar007)
local_gov_indicators_2019$itpcm <- as.numeric(local_gov_indicators_2019$itpcm)
local_gov_indicators_2019$itpc <- as.numeric(local_gov_indicators_2019$itpc)
local_gov_indicators_2019$itpr <- as.numeric(local_gov_indicators_2019$itpr)
local_gov_indicators_2019$itpu <- as.numeric(local_gov_indicators_2019$itpu)

## SUBTEL Data
local_gov_indicators_2019 <- left_join(local_gov_indicators_2019, subtel_2019[c(2, 15)], by = "cut_com")
local_gov_indicators_2019$internet_con <- as.numeric(local_gov_indicators_2019$internet_con)

## Case-Level CSV
write.csv(local_gov_indicators_2019, "data/tidy/cases/local_gov_indicators_2019.csv", 
          fileEncoding = "UTF-8", row.names =  FALSE)

## +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

#### Case Level-Data Set 2021 ####

## +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

## Subsample
## table(egi_2021$region)
subsample_2021 <- filter(egi_2021, region == "COQUIMBO" | region == "VALPARAISO" 
                         | region == "METROPOLITANA" | region == "BIOBIO" | region == "DE LA ARAUCANIA")

## Alpha
## Incorporating digital_certificate
alpha_egi_2021 <- select(subsample_2021, forms, news, webmap, streetmap, transport, socialmedia, phone, 
                         mobile, transac, followup, digital_certificate, citizen, payments, part)
## alpha(alpha_egi_2021)
ltm::cronbach.alpha(alpha_egi_2021, CI = TRUE, na.rm = TRUE)

## EGI
egi_2021$egi <- ((egi_2021$forms + egi_2021$news + egi_2021$webmap) * 0.25) + 
  ((egi_2021$streetmap + egi_2021$transport) * 0.50) +
  ((egi_2021$socialmedia + egi_2021$phone + egi_2021$mobile) * 0.75) +
  ((egi_2021$transac + egi_2021$followup + egi_2021$digital_certificate + egi_2021$citizen + egi_2021$payments) * 1.00) +
  ((egi_2021$part) * 1.25)
sum(egi_2021$egi, na.rm = TRUE)

## Local Gov Indicators
local_gov_indicators_2021 <- cut_shapefiles
local_gov_indicators_2021 <- cbind(year = 2021, local_gov_indicators_2021)
local_gov_indicators_2021 <- cbind(id = rownames(local_gov_indicators_2021), local_gov_indicators_2021)

## Merge
local_gov_indicators_2021 <- left_join(local_gov_indicators_2021, egi_2021[c("cut_com", "egi")], by = "cut_com")
sum(local_gov_indicators_2021$egi, na.rm = TRUE)
local_gov_indicators_2021$egi_std <- NA

## CPLT Data
local_gov_indicators_2021$ta <- NA
local_gov_indicators_2021$dai <- NA

## SAE Data
local_gov_indicators_2021$sae <- NA

## SINIM Data
local_gov_indicators_2021 <- left_join(local_gov_indicators_2021, sinim_2021, by = "cut_com")
local_gov_indicators_2021$municipality <- NULL
local_gov_indicators_2021$iadm40 <- as.numeric(local_gov_indicators_2021$iadm40)
local_gov_indicators_2021$iadm44 <- as.numeric(local_gov_indicators_2021$iadm44)
local_gov_indicators_2021$iadm41 <- as.numeric(local_gov_indicators_2021$iadm41)
local_gov_indicators_2021$iadm25 <- as.numeric(local_gov_indicators_2021$iadm25)
local_gov_indicators_2021$iadm33 <- as.numeric(local_gov_indicators_2021$iadm33)
local_gov_indicators_2021$isoc013 <- as.numeric(local_gov_indicators_2021$isoc013)
local_gov_indicators_2021$mocosoc <- as.factor(local_gov_indicators_2021$mocosoc)
local_gov_indicators_2021$mcosoc <- as.factor(local_gov_indicators_2021$mcosoc)
local_gov_indicators_2021$icar007 <- as.numeric(local_gov_indicators_2021$icar007)
local_gov_indicators_2021$itpcm <- as.numeric(local_gov_indicators_2021$itpcm)
local_gov_indicators_2021$itpc <- as.numeric(local_gov_indicators_2021$itpc)
local_gov_indicators_2021$itpr <- as.numeric(local_gov_indicators_2021$itpr)
local_gov_indicators_2021$itpu <- as.numeric(local_gov_indicators_2021$itpu)

## SUBTEL Data
local_gov_indicators_2021 <- left_join(local_gov_indicators_2021, subtel_2021[c(2, 15)], by = "cut_com")
local_gov_indicators_2021$internet_con <- as.numeric(local_gov_indicators_2021$internet_con)

## Case-Level CSV
write.csv(local_gov_indicators_2021, "data/tidy/cases/local_gov_indicators_2021.csv", 
          fileEncoding = "UTF-8", row.names =  FALSE)

## +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

#### Scatters ####

## +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

library(ggplot2)

df_scatter <- data.frame(cut_com = local_gov_indicators_2016$cut_com,
                         egi_2016 = local_gov_indicators_2016$egi_std,
                         egi_2019 = local_gov_indicators_2019$egi,
                         egi_2021 = local_gov_indicators_2021$egi)

summary(lm(egi_2019 ~ egi_2016, data = df_scatter))
summary(lm(egi_2021 ~ egi_2019, data = df_scatter))

## pdf("results/figures/scatter_egi_2016_2019.pdf", width = 6.826666666666667, height = 5.12)
## png("results/figures/scatter_egi_2016_2019.png", width = (1024*2), height = (768*2), units = 'px', res = 300)
ggplot(df_scatter, aes(x = egi_2016, y = egi_2019)) + 
  geom_point(alpha = 0.15) + 
  ## geom_smooth(method = "loess", color = "tomato2", fill = "tomato1") +
  geom_smooth(method = lm, colour = "grey30", alpha = 0.25) +
  theme_minimal(base_size = 12) + theme(legend.position = "bottom") +
  theme(panel.grid.minor = element_blank()) +
  theme(axis.text.x = element_text(angle = 35, hjust = 1, color = "black", size = 9)) +
  labs(x = "EGI 2016", y = "EGI 2019", title = "E-Government Index", subtitle = "Chilean Municipalities", 
       colour = NULL, caption = "Source: González-Bustamante and Aguilar (2022).") +
  theme(plot.margin = unit(c(0.5,0.5,0.5,0.5), "cm")) 
## dev.off()

## pdf("results/figures/scatter_egi_2019_2021.pdf", width = 6.826666666666667, height = 5.12)
## png("results/figures/scatter_egi_2019_2021.png", width = (1024*2), height = (768*2), units = 'px', res = 300)
ggplot(df_scatter, aes(x = egi_2019, y = egi_2021)) + 
  geom_point(alpha = 0.15) + 
  ## geom_smooth(method = "loess", color = "tomato2", fill = "tomato1") +
  geom_smooth(method = lm, colour = "grey30", alpha = 0.25) +
  theme_minimal(base_size = 12) + theme(legend.position = "bottom") +
  theme(panel.grid.minor = element_blank()) +
  theme(axis.text.x = element_text(angle = 35, hjust = 1, color = "black", size = 9)) +
  labs(x = "EGI 2019", y = "EGI 2021", title = "E-Government Index", subtitle = "Chilean Municipalities", 
       colour = NULL, caption = "Source: González-Bustamante and Aguilar (2022).") +
  theme(plot.margin = unit(c(0.5,0.5,0.5,0.5), "cm")) 
## dev.off()
